vowpalwabbit/cb.h

```diff
@@ -5,23 +5,35 @@
 
 #include <vector>
 #include <cstdint>
+#include <utility>
 
 #include "reductions_fwd.h"
 #include "label_parser.h"
 #include "v_array.h"
 
 struct example;
+using multi_ex = std::vector<example*>;
 
 namespace CB
 {
+// By default a cb class does not contain an observed cost.
 struct cb_class
 {
-  float cost;         // the cost of this class
-  uint32_t action;    // the index of this class
-  float probability;  // new for bandit setting, specifies the probability the data collection policy chose this class
-                      // for importance weighting
-  float partial_prediction;  // essentially a return value
-  bool operator==(cb_class j) { return action == j.action; }
+  float cost = FLT_MAX;      // the cost of this class
+  uint32_t action = 0;       // the index of this class
+  float probability = -1.f;  // new for bandit setting, specifies the probability the data collection policy chose this
+                             // class for importance weighting
+  float partial_prediction = 0.f;  // essentially a return value
+
+  cb_class() = default;
+  cb_class(float cost, uint32_t action, float probability)

```

Comment:
It seems dangerous having both using list initialization and having a constructor like this. If the member variables aren't ordered the same way, then their meanings will be different, which can lead to some hard to debug issues

Reply:
Where is the list initialization?

Reply:
every usage of cb_class is done via list initialization..  the first diff on the PR for example `ec.l.cb.costs.push_back(CB::cb_class{3.5f, 2, 0.5f});`

Reply:
`CB::cb_class{3.5f, 2, 0.5f}` is calling this constructor. List initialization is not available for this type now. (It used to be but this removes that)